lib/flat_search_web/live/page_live.ex

```diff
@@ -18,6 +18,78 @@ defmodule FlatSearchWeb.PageLive do
       |> Filters.change_filter(params)
       |> Map.put(:action, :insert)
 
-    {:noreply, assign(socket, changeset: changeset)}
+    {:noreply,
+     assign(socket, changeset: changeset, flats: Flats.get_flats_by(params), filters: params)}
+  end
+
+  @impl true
+  def handle_info({:flat_created, flat}, socket) do
+    insensitive_string("Warszawą Dąbrowa")
+
+    case flat_fulfills_filters?(socket.assigns.filters, flat) do
+      true ->
+        {:noreply, assign(socket, flats: [flat] ++ socket.assigns.flats)}
+
+      false ->
+        {:noreply, assign(socket, flats: socket.assigns.flats)}
+    end
+  end
+
+  @impl true
+  def handle_info({:error, reason}, socket) do
+    {:noreply, assign(socket, reason: reason)}
+  end
+
+  defp flat_fulfills_filters?(user_filters, flat_filters) do
+    # Compare only if:
+    # - key in flat_filters exist
+    # - value that key points on in user_filters is not empty
+
+    # support a case for price
+    key_exists_and_has_value =
+      for {key, value} when value not in ["", nil] <- user_filters do
+        case key do
+          "max_price" ->
+            Map.has_key?(flat_filters, :price) and Map.has_key?(flat_filters, :additional_price)
+
+          key ->
+            Map.has_key?(flat_filters, String.to_existing_atom(key))
+        end
+      end
+
+    case check_lsit_truthy(key_exists_and_has_value) do
+      false ->
+        false
+
+      true ->
+        for {key, value} when value not in ["", nil] <- user_filters do
+          case key do
+            "max_price" ->
+              String.to_integer(value) >= flat_filters.price + flat_filters.additional_price
+
+            _ ->
+              insensitive_string(flat_filters[String.to_existing_atom(key)]) ==
+                insensitive_string(value)
+          end
+        end
+        |> check_lsit_truthy()
+    end
+  end
+
+  defp check_lsit_truthy(list) do
```

Comment:
typo :P

Reply:
Also maybe just  as a function body
`false not in list`

Reply:
What do you mean?

Reply:
the inside of this function

```
    cond do
      false in list ->
        false

      false not in list ->
        true
        # not a list case
    end
 ```

Reply:
just `false not in list` will return true if `false` is not in the `list`, or true if `false` is in the `list`

Reply:
Oh... right - so the whole cond is unnecessary